model/src/main/java/com/flipkart/flux/domain/State.java

```diff
@@ -47,7 +47,7 @@
     /** Description for this State*/
     private String description;
     /** Id of the state machine to which this state belongs*/
-    private Long stateMachineId;
+    private String stateMachineId;
```

Comment:
As discussed earlier, let's have <stateMachineId, taskId> as primary key for State table